lib/platform/github/index.ts

```diff
@@ -307,16 +307,40 @@ export async function initRepo({
         token: forkToken || opts.token,
       });
       config.repository = forkedRepo.body.full_name;
-      config.forkDefaultBranch = forkedRepo.body.default_branch;
+      const forkDefaultBranch = forkedRepo.body.default_branch;
       // istanbul ignore if
-      if (config.forkDefaultBranch !== config.defaultBranch) {
+      if (forkDefaultBranch !== config.defaultBranch) {
+        const body = {
+          ref: `refs/heads/config.defaultBranch`,
+          sha: repo.defaultBranchRef.target.oid,
+        };
         logger.debug(
           {
             defaultBranch: config.defaultBranch,
-            forkDefaultBranch: config.forkDefaultBranch,
+            forkDefaultBranch,
+            body,
           },
-          'Fork has different default branch to parent'
+          'Fork has different default branch to parent, attempting to create branch'
         );
+        try {
+          await githubApi.postJson(`repos/${config.repository}/git/refs`, {
+            token: forkToken || opts.token,
+          });
+          logger.debug('Created new default branch in fork');
+        } catch (err) {
+          logger.warn({ err }, 'Could not create parent defaultBranch in fork');
```

Comment:
This is not unrecoverable - it could be because the branch somehow already existed